### ISSUE TITLE
Not able to reproduce same data from dataloader in pytorch

### ISSUE THREAD

**Issue by deeppatel4557** [pytorch/pytorch#39571]
If you have a question or would like help and support, please ask at our
[forums](https://discuss.pytorch.org/).

If you are submitting a feature request, please preface the title with [feature request].
If you are submitting a bug report, please fill in the following details.

## Issue description
I am not able to reproduce same result when dataloader is executing again my shuffle value is true
Provide a short description.

## Code example
def _init_fn():
    SEED=4
    torch.manual_seed(SEED)
    torch.cuda.manual_seed(SEED)
    np.random.seed(SEED)
    random.seed(SEED)
    torch.backends.cudnn.deterministic=True
trainloader = torch.utils.data.DataLoader(train_dataset, batch_size=32, shuffle=True,worker_init_fn=_init_fn)
Please try to provide a minimal example to repro the bug.
Error messages and stack traces are also helpful.

## System Info
Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

- PyTorch or Caffe2:
- How you installed PyTorch (conda, pip, source):
- Build command you used (if compiling from source):
- OS:
- PyTorch version:
- Python version:
- CUDA/cuDNN version:
- GPU models and configuration:
- GCC version (if compiling from source):
- CMake version:
- Versions of any other relevant libraries:

**Comment by deeppatel4557** [comment]
I think running init_fn whenever we iterate through dataloader works!!!!!
Please tell me if this works for someone as well.
